v2/internal/carbon/carbon_fds.go

```diff
@@ -14,8 +15,8 @@ import (
 type carbonFD struct {
 	path        string
 	writeHandle *poswriter
-	carbs.BlockStore
-	idx *insertionIndex
+	blockstore.ReadOnlyBlockStore
```

Comment:
like the filename, I wonder if this type stutters a bit. `blockstore.ReadOnly` is pretty good, or if you still prefer to clarify what it is, maybe `blockstore.ReadOnlyStore`?

also, nit: I'm pretty sure the go-ipfs-blockstore package uses the name Blockstore, not BlockStore (note the lack of capitalization of Store), so I think go-car should be consistent too. Maybe its name is a historical typo/mistake, but best to not confuse things further with multiple spellings.